packages/react/src/hooks/use-element-rect.ts

```diff
@@ -1,29 +1,44 @@
 "use client"
 
 import { useEnvironmentContext } from "@ark-ui/react"
-import { useEffect, useRef, useState } from "react"
+import { useCallback, useEffect, useRef, useState } from "react"
 
 export function useElementRect() {
   const ref = useRef<HTMLElement | null>(null)
   const [rect, setRect] = useState<DOMRect | null>(null)
   const env = useEnvironmentContext()
 
+  // Extract the calculation logic to a callback
+  const updateRect = useCallback(() => {
```

Comment:
This seems wrong. We added extra work and didn't solve anything. I would revert this change back to `handleResize`. 
Real solution would be [`useEffectEvent`](https://react.dev/reference/react/experimental_useEffectEvent) but it's not released yet.
`useCallback` is meant to only be used for optimising callback props to prevent component re-rendering (in combination with `memo`), not to be used inside `useEffects`. This could easily get unoptimised if someone accidentally add dependency to the array in the future update.